### PR TITLE
Add fluidName() method to fluid_properties

### LINKED ISSUE #9094: FluidProperties: add fluidName() method
### Description of the enhancement or error report
Add a fluidName() method to fluids that returns a string that is the name of the fluid.

### Rationale for the enhancement or information for reproducing the error
To use in the construction of some complex `porous_flow` objects that are only really applicable for specific fluids to check for accidental errors in the input file that will result in a simulation that runs but gives the incorrect results, eg. check that the `fluid_properties` UserObject that should be water is water and throw an error if it isn't.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
Enables error checking during construction to catch errors in the input files

### PR DESCRIPTION
**Pull request by cpgr**
Closes #9094

### CODE REVIEW

**Comment by cpgr** [comment]
Of course, calling this member `name()` wasn't such a good idea...

**Comment by permcody** [comment]
What if you could retrieve the stringified type itself from the base class? We could make that available. I'm fine merging this in, just want to make sure you've chosen the right design before clicking the big green button.

**Comment by andrsd** [comment]
That would not work for the case where CO2 properties are provided by either `TabulatedFluidProperties` or `CO2FluidProperties`. This setup (specific class vs tabulated version) can be used for any fluid...

**Comment by permcody** [comment]
@andrsd - feel free to merge if you think this is fine.

File: modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C [hunk 1]
```diff
@@ -108,6 +108,13 @@ TabulatedFluidProperties::initialSetup()
       _pressure, _temperature, _enthalpy, _dh_dp_0, _dh_dp_n, _dh_dT_0, _dh_dT_n);
 }
 
+std::string
+
```
Comment (andrsd) [inline modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C]: And `clang-format` did not remove this empty line?

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C (hunk 1 above)
Comment (cpgr) [inline modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C]: Apparently not.

**Review by andrsd — APPROVED** [review]

**Pull request merged**